internal/server/spanner/query.go

```diff
@@ -18,25 +18,86 @@ package spanner
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"cloud.google.com/go/spanner"
+	"github.com/datacommonsorg/mixer/internal/server/v2/shared"
 	"google.golang.org/api/iterator"
 )
 
+var (
+	ObsColumns = []string{
+		"variable_measured",
+		"observation_about",
+		"observation_date",
+		"value",
+		"provenance",
+		"observation_period",
+		"measurement_method",
+		"unit",
+		"scaling_factor",
+	}
+)
+
 // SQL / GQL statements executed by the SpannerClient
 var statements = struct {
-	getEdgesBySubjectID string
+	getEdgesBySubjectID             string
+	getObsByVariableAndEntity       string
+	getObsByVariableEntityAndDate   string
+	getLatestObsByVariableAndEntity string
 }{
 	getEdgesBySubjectID: `
-	SELECT 
-		subject_id, 
-		predicate, 
-		COALESCE(object_id, '') AS object_id, 
-		COALESCE(object_value, '') AS object_value, 
-		COALESCE(provenance, '') AS provenance
-	FROM Edge
-	WHERE subject_id IN UNNEST(@ids)
+		SELECT 
+			subject_id, 
+			predicate, 
+			COALESCE(object_id, '') AS object_id, 
+			COALESCE(object_value, '') AS object_value, 
+			COALESCE(provenance, '') AS provenance
+		FROM Edge
+		WHERE subject_id IN UNNEST(@ids)
+	`,
+	getObsByVariableAndEntity: fmt.Sprintf(`
+		SELECT %s
+		FROM StatVarObservation
+		WHERE
+			variable_measured IN UNNEST(@variables) AND
+			observation_about IN UNNEST(@entities) AND
+			value != ''
+		ORDER BY observation_date ASC
 	`,
+		getSelectColumns(ObsColumns, "")),
+	getObsByVariableEntityAndDate: fmt.Sprintf(`
+		SELECT %s
+		FROM StatVarObservation
+		WHERE
+			variable_measured IN UNNEST(@variables) AND
+			observation_about IN UNNEST(@entities) AND
+			observation_date = @date AND
+			value != ''
+		ORDER BY observation_date ASC
+	`,
+		getSelectColumns(ObsColumns, "")),
+	getLatestObsByVariableAndEntity: fmt.Sprintf(`
```

Comment:
I think this will return the latest observation overall, rather than per facet

Reply:
hmm - true. So in that case, we probably fetch them all from the DB and then process in mixer to return the latest per facet?

Reply:
That could work, or we could group by facet in the query

Reply:
Let me try the latter. Will fetch less data.

Reply:
Done. Let me know if this looks ok.